01-JS-DOM-APIs/Practice1-10/Script/script.js

```diff
@@ -0,0 +1,48 @@
+function load () {
+  console.log('load'); //Only to know the page is loaded
+  let changeElement = document.getElementById("changeclass");
+  changeElement.style.transition = "all 3s ease-in-out";
+  changeElement.style.opacity = 1;
+}
+
+function configAjax (methodHttp, url, asyncronic){
+  this.methodHttp = methodHttp;
+  this.url = url;
+  this.asyncronic = asyncronic;
+}
+
+function request(objconfig) {
+  return new Promise(function (resolve, reject) {
+    let req = new XMLHttpRequest();
+    req.open(objconfig.methodHttp, objconfig.url);
+    req.onload = function() {
+     if (req.status == 200) {
+      resolve(req.response);
+     }
+     else {
+      reject(Error(req.statusText));
+     }
+    };
+    req.onerror = function() {
+      reject(Error('Network Error'));
+    };
+    req.send();
+  });
+}
+
+function connect(inpText){
+    let objconfig = new configAjax("GET", "https://api.github.com/search/repositories?q="+inpText, true);
+    request(objconfig).then(function(response) {
+     resp = JSON.parse(response);
+     console.log(resp);
+     for (var key = 0; key < resp.items.length; key++){
+      var columnNode = document.createElement("li");
```

Comment:
Well, i commit whit some error, because when i declare the variable outside the loops i got trouble whit the textNode( ).
I fixed the error later.